src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkCapacity();
+        int index = getIndex(key);
+        if (buckets[index] == null) {
+            Node<K, V> newNode = new Node<>(key, value);
+            buckets[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        newNode.next = buckets[index];
+        buckets[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkCapacity() {
+        if (size == buckets.length * DEFAULT_LOAD_FACTOR) {
+            increaseCapacity();
+        }
+    }
+
+    private void increaseCapacity() {
+        int newCapacity = buckets.length * RESIZE_FACTOR;
+        Node<K, V>[] oldBuckets = buckets;
+        buckets = (Node<K, V>[]) new Node[newCapacity];
+        size = 0;
+        for (Node<K, V> node : oldBuckets) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode()) % buckets.length;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
```

Comment:
better to initialize all fields  in constructor (not in general but in this case)